common/scenario_simulator_exception/include/scenario_simulator_exception/exception.hpp

```diff
@@ -51,10 +53,28 @@ DEFINE_ERROR_CATEGORY(SemanticError);
 // A problem occurred that interfered with the continuation of the simulation.
 DEFINE_ERROR_CATEGORY(SimulationError);
 
+// Metric module detects specification violation in simulation.
+DEFINE_ERROR_CATEGORY(SpecificationViolation);
+
 // There is a syntactic error in the description of the scenario. Or you are
 // using a feature that is not yet supported by our implementation.
 DEFINE_ERROR_CATEGORY(SyntaxError);
 
+#define THROW_AUTOWARE_ERROR(...) \
+  THROW_ERROR(common::scenario_simulator_exception::AutowareError, __VA_ARGS__);
+#define THROW_SEMANTIC_ERROR(...) \
+  THROW_ERROR(common::scenario_simulator_exception::SemanticError, __VA_ARGS__);
+#define THROW_SIMULATION_ERROR(...) \
+  THROW_ERROR(common::scenario_simulator_exception::SimulationError, __VA_ARGS__);
+#define THROW_SPECIFICATION_VIOLATION(...) \
+  THROW_ERROR(common::scenario_simulator_exception::SpecificationViolation, __VA_ARGS__);
+#define THROW_SYNTAX_ERROR(...) \
+  THROW_ERROR(common::scenario_simulator_exception::SyntaxError, __VA_ARGS__);
```

Comment:
Semicolons should be added by the macro user and should not be included in the macro itself.

Reply:
Thank you for pointing out! I will fix soon.

Reply:
fixed in https://github.com/tier4/scenario_simulator_v2/pull/306/commits/4e2999f700d892c18aaca045592630dc006ef69b